src/common/AndroidEnvironmentRequirements.ts

```diff
@@ -16,27 +17,76 @@ const messages = Messages.loadMessages(
     'requirement-android'
 );
 
-export class AndroidEnvironmentRequirements implements RequirementList {
+export class AndroidSDKRootSetRequirements implements RequirementList {
     public requirements: Requirement[] = [];
     public enabled = true;
-    constructor(logger: Logger, apiLevel?: string) {
+    public title = 'Android SDK Requirements';
+    constructor(logger: Logger) {
         this.requirements = [
             new AndroidSDKRootSetRequirement(logger),
-            new Java8AvailableRequirement(logger),
-            new AndroidSDKToolsInstalledRequirement(logger),
-            new AndroidSDKPlatformToolsInstalledRequirement(logger),
-            new PlatformAPIPackageRequirement(logger, apiLevel),
-            new EmulatorImagesRequirement(logger, apiLevel)
+            new AndroidSDKInstallTask(logger)
+        ];
+    }
+}
+
+export class Java8AvailableRequirements implements RequirementList {
+    public requirements: Requirement[] = [];
+    public enabled = true;
+    public title = 'Java 8 Requirements';
+    constructor(logger: Logger) {
+        this.requirements = [new Java8AvailableRequirement(logger)];
```

Comment:
Why did you decide to create these extra classes as `RequirementList` but with only 1 requirement in the list? Couldn't we just stick to the original class extending `Requirement`?

Reply:
This is the anticipation of creating the install tasks for each in the future similar to the SDK root task.